### PR TITLE
ASP-2712: Add changes to support slurm 22.05.8

### PR DESCRIPTION
**Pull request by jaimesouza**
What

Add changes to support Slurm 22.05.8.

Why

Slurm will be upgraded to version 22.05.8.

### CODE REVIEW

**Review by matheushent — COMMENTED** [review]
why can't the `slurm` user be used by the slurmrestd process?

**Comment by jaimesouza** [comment]
> why can't the `slurm` user be used by the slurmrestd process?

According to https://slurm.schedmd.com/rest.html, while using JWT authentication, slurmrestd should be run as a unique, unprivileged user and group.

File: slurm_ops_manager/slurm_ops_base.py [hunk 1]
```diff
@@ -260,11 +260,69 @@ def _slurmd_group(self) -> str:
...
+                return False
+
+        try:
+            subprocess.check_output([
+                "adduser",
+                "--system",
+                "--gid", self._slurmrestd_group_id,
+                "--uid", self._slurmrestd_user_id,
+                "--no-create-home",
+                "--home", "/nonexistent",
```
Comment (matheushent) [inline slurm_ops_manager/slurm_ops_base.py:314]: is this really correct?

Comment (jaimesouza) [reply slurm_ops_manager/slurm_ops_base.py:314]: Yes, it is.

Without` --home /nonexistent`
```
$ sudo adduser --system --gid 64031 --no-create-home someuser

$ cat /etc/passwd
someuser:x:996:64031::/home/someuser:/bin/bash
```

It does not create the home dir, but it still sets it in the user info.

With `--home /nonexistent`

```
$ sudo adduser --system --gid 64031 --no-create-home --home /nonexistent someuser2

$ cat /etc/passwd
someuser2:x:995:64031::/nonexistent:/bin/bash
```

It enforces using the /nonexistent in the user info. 

adduser --system should default to /nonexistent, but it isn't.

According to https://www.debian.org/doc/debian-policy/ch-opersys.html#non-existent-home-directories, the canonical non-existent home directory is /nonexistent. Users who should not have a home directory should have their home directory set to this value.

File: slurm_ops_manager/slurm_ops_base.py [hunk 2]
```diff
@@ -260,11 +260,69 @@ def _slurmd_group(self) -> str:
         """Return the slurmd group."""
         return "root"
 
+    @property
+    def _slurmrestd_user_id(self) -> str:
+        """Return the slurmrestd user ID."""
+        return "64031"
```
Comment (matheushent) [inline slurm_ops_manager/slurm_ops_base.py:266]: just to confirm: does slurm 22.05.8 require this specific ID or did you just pick one that it's not intuitive to exist?

**Pull request merged**